File: 其他/LeetCode算法/calcu/2020.8 - now/10.12/剑指 Offer 05. 替换空格.py
class Solution:
    def replaceSpace(self, s: str) -> str:
        return s.replace(' ', '%20')
        # ans = ''
        # for c in s :
        #     ans += c if c != ' ' else '%20'
        # return ans

solute = Solution()
print(solute.replaceSpace('We are happy.'))

File: 其他/LeetCode算法/calcu/2020.8 - 6.15/10.16/剑指 Offer 05. 替换空格.js
  /**
 * @param {string} s
 * @return {string}
 */
var replaceSpace = function(s) {
  newStr = s.replace(/\s/g, '%20')
  return newStr
  let ans = ''
  for (var c of s) {
    if (c === ' ') {
      ans += '%20'
    } else {
      ans += c
    }
  }
  return ans
};

console.log(replaceSpace('We are happy.'))